fastcrypto-tbls/src/dkg.rs

```diff
@@ -73,13 +75,47 @@ pub struct ProcessedMessage<G: GroupElement, EG: GroupElement> {
 /// Mapping from node id to the shares received from that sender.
 pub type SharesMap<S> = HashMap<PartyId, Vec<Share<S>>>;
 
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct UsedProcessedMessages<G: GroupElement, EG: GroupElement>(
+    pub Vec<ProcessedMessage<G, EG>>,
+);
+
+impl<G: GroupElement, EG: GroupElement> From<&[ProcessedMessage<G, EG>]>
+    for UsedProcessedMessages<G, EG>
+{
+    fn from(msgs: &[ProcessedMessage<G, EG>]) -> Self {
+        let filtered = msgs
+            .iter()
+            .unique_by(|&m| m.message.sender)
+            .cloned()
+            .collect::<Vec<_>>();
+        Self(filtered)
+    }
+}
+
+pub struct VerifiedProcessedMessages<G: GroupElement, EG: GroupElement>(
+    pub Vec<ProcessedMessage<G, EG>>,
+);
+
+impl<G: GroupElement, EG: GroupElement> VerifiedProcessedMessages<G, EG> {
+    fn filter_from(msgs: &UsedProcessedMessages<G, EG>, to_exclude: &[PartyId]) -> Self {
+        let filtered = msgs
+            .0
+            .iter()
+            .filter(|m| !to_exclude.contains(&m.message.sender))
+            .cloned()
+            .collect::<Vec<_>>();
+        Self(filtered)
+    }
+}
+
 /// [Output] is the final output of the DKG protocol in case it runs
 /// successfully. It can be used later with [ThresholdBls], see examples in tests.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Output<G: GroupElement, EG: GroupElement> {
     pub nodes: Nodes<EG>,
     pub vss_pk: Poly<G>,
-    pub shares: Vec<Share<G::ScalarType>>,
+    pub shares: Option<Vec<Share<G::ScalarType>>>, // None if some shares are missing.
```

Comment:
Maybe add a bit more to the doc comment about what to do in case this is None? I guess then you can still use this to verify full sigs, but can't generate partial sigs?